gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -581,6 +655,9 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
                 (("resource", request.resource),)),
         )
 
+        # Validate the universe domain.
+        self.transport._credentials and self._validate_universe_domain()
```

Comment:
Let's move the `self.transport._credentials` check into `_validate_universe_domain`, with something in that method like

```
self._is_universe_domain_valid = ( self.transport._credentials is None or
                                   ConfigServiceV2Client._compare_universes(self.universe_domain, self.transport._credentials.universe_domain)
```


and then this line here becomes just the function call

Reply:
addressed.